pmd-vm/src/main/java/net/sourceforge/pmd/lang/vm/ast/TokenMgrError.java

```diff
@@ -101,7 +101,7 @@ protected static final String addEscapes(final String str) {
                 ch = str.charAt(i);
                 if (ch < 0x20 || ch > 0x7e) {
                     final String s = "0000" + Integer.toString(ch, 16);
-                    retval.append("\\u" + s.substring(s.length() - 4, s.length()));
+                    retval.append("\\u").append(s.substring(s.length() - 4, s.length()));
```

Comment:
Please don't edit this file as it has been removed in another branch

Reply:
@oowekyala reverted&rebased